src/crimson/os/seastore/transaction_manager.h

```diff
@@ -857,18 +864,33 @@ class TransactionManager : public ExtentCallbackInterface {
       pref.is_indirect() ?
 	pref.get_intermediate_length() :
 	pref.get_length(),
-      [pin=std::move(pin)]
+      [&pref]
       (T &extent) mutable {
 	assert(!extent.has_laddr());
 	assert(!extent.has_been_invalidated());
-	assert(!pin->has_been_invalidated());
-	assert(pin->get_parent());
-	pin->link_child(&extent);
-	extent.maybe_set_intermediate_laddr(*pin);
+	assert(!pref.has_been_invalidated());
+	assert(pref.get_parent());
+	pref.link_child(&extent);
+	extent.maybe_set_intermediate_laddr(pref);
       }
-    ).si_then([FNAME, &t](auto ref) mutable -> ret {
-      SUBTRACET(seastore_tm, "got extent -- {}", t, *ref);
+    ).si_then([FNAME, &t, pin=std::move(pin), this](auto ref) mutable -> ret {
+      auto crc = ref->calc_crc32c();
+      SUBTRACET(
+	seastore_tm,
+	"got extent -- {}, chksum in the lba tree: {}, actual chksum: {}",
+	t,
+	*ref,
+	pin->get_checksum(),
+	crc);
       assert(ref->is_fully_loaded());
+      if (pin->is_indirect()) {
+	ceph_assert(pin->get_checksum() == 0);
+      } else if (full_extent_integrity_check) {
+	ceph_assert(pin->get_checksum() == crc);
+      } else { // !full_extent_integrity_check: remapped extent may be skipped
+	ceph_assert(pin->get_checksum() == 0 ||
+		    pin->get_checksum() == crc);
+      }
```

Comment:
Checksum mismatch may not be logical bugs -- it may caused by device issues, and is even possible to be recoverred -- so print how the information is mismatched as error log before the according assert failure (or abort) is necessary here, I think.